pwa/app/components/tba/eliminationBracket.tsx

```diff
@@ -0,0 +1,697 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router';
+
+import PlayCircle from '~icons/bi/play-circle';
+
+import { EliminationAlliance, Event, Match } from '~/api/tba/read';
+import { TeamLink } from '~/components/tba/links';
+import { EventType } from '~/lib/api/EventType';
+
+export interface EliminationBracketProps {
+  alliances: EliminationAlliance[];
+  matches: Match[];
+  year?: number;
+  event?: Event;
+}
+
+export default function EliminationBracket({
+  alliances,
+  matches,
+  year,
+  event,
+}: EliminationBracketProps): React.JSX.Element {
+  const [hoveredAlliance, setHoveredAlliance] = useState<number | null>(null);
+
+  // Check if this is an Einstein event (CMP_FINALS)
+  const isEinsteinEvent = event?.event_type === EventType.CMP_FINALS;
+
+  // Helper to get division shortform (e.g., "Newton Division" -> "New")
+  const getDivisionShortform = (divisionName: string): string => {
+    // Common division names and their shortforms
+    const divisionShortforms: Record<string, string> = {
+      Newton: 'New',
+      Einstein: 'Ein',
+      Curie: 'Cur',
+      Galileo: 'Gal',
+      Hopper: 'Hop',
+      Tesla: 'Tes',
+      Turing: 'Tur',
+      Archimedes: 'Arc',
+      Carson: 'Car',
+      Carver: 'Crv',
+      Daly: 'Dal',
+      Darwin: 'Dar',
+      Johnson: 'Joh',
+      Milstein: 'Mil',
+      Roebling: 'Roe',
+    };
+
+    // Try to match the division name
+    for (const [fullName, shortForm] of Object.entries(divisionShortforms)) {
+      if (divisionName.includes(fullName)) {
+        return shortForm;
+      }
+    }
+
+    // If no match found, take first 3 letters
+    return divisionName.substring(0, 3);
+  };
+
+  // Helper to get alliance display name
+  const getAllianceDisplayName = (allianceNumber: number): string => {
+    if (!allianceNumber || allianceNumber > alliances.length) return '';
+
+    const alliance = alliances[allianceNumber - 1];
+    if (isEinsteinEvent && alliance.name) {
+      // For Einstein events, use division shortform if available
+      return getDivisionShortform(alliance.name);
+    }
+    // Default to alliance number
+    return `#${allianceNumber}`;
+  };
+
+  if (
+    !alliances ||
+    alliances.length === 0 ||
+    !matches ||
+    matches.length === 0
+  ) {
+    return <div></div>;
+  }
+
+  // Group matches by set_number
+  const matchesBySet = matches.reduce(
```

Comment:
You might be able to simplify this with lodash-es. Try searching for some method in here: https://lodash.com/docs/

If not, no big deal though